sabnzbd/deobfuscate_filenames.py

```diff
@@ -225,90 +233,151 @@ def deobfuscate(nzo, filelist: List[str], usefulname: str):
-            # Rename and make sure the new filename is unique
-            renamer(otherfile, new_name)
+            filtered_filelist.remove(otherfile)
+            filtered_filelist.append(renamer(otherfile, new_name))
             nr_files_renamed += 1
 
     if nr_files_renamed:
         nzo.set_unpack_info("Deobfuscate", T("Deobfuscate renamed %d file(s)") % nr_files_renamed)
+
+    return filtered_filelist
+
+
+def without_extension(fullpathfilename):
+    """Returns full file path, without extension
+    So '/some/dir/somefile.bin' results in '/some/dir/somefile'"""
+    return os.path.splitext(fullpathfilename)[0]
+
+
+def deobfuscate_subtitles(input):
+    """
+    input can be an existing directory or a list of existing filenames
+
+    Find .srt subtitle files, and rename to match the biggest file (if there is a clearly biggest file)
+
+    Some_Big_File_2024.mp4      # biggest file
+    Some_Big_File_2024.srt      # no renaming wanted
+    Some_Big_File_2024.ger.srt  # no renaming wanted
+    14_English.srt              # to be renamed
+    dut.srt                     # to be renamed
+    Something.else.txt          # no renaming wanted, because no .srt
+
+    will result in
+
+    Some_Big_File_2024.mp4
+    Some_Big_File_2024.srt
+    Some_Big_File_2024.ger.srt
+    Some_Big_File_2024.14.English.srt   # renamed by prepending base name (and replacing _)
+    Some_Big_File_2024.dut.srt          # renamed by prepending base name
+    Something.else.txt
+
+    """
+
+    # if directory, create list from that directory
+    if type(input) == str and os.path.isdir(input):
+        all_files = [os.path.join(root, file) for root, dirs, files in os.walk(input) for file in files]
+    elif type(input) == list:
+        # already a list, so nothing to do
+        all_files = input
+    else:
+        logging.debug(f"{input} no directory and no list")
+        return None
+
+    # find .srt files
+    if not (srt_files := [f for f in all_files if f.endswith(".srt")]):
+        logging.debug("No .srt files found, so nothing to do")
+        return None
+
+    # check there is a clearly biggest file
+    if not (biggest_file := get_biggest_file(all_files)):
+        logging.debug("No clearly biggest file found, so no subtitle renaming feasible")
+        return None
+
+    biggest_file_without_ext = without_extension(biggest_file)  # get full path base name of biggest file
+    logging.debug(f"biggest_file {biggest_file} with base {biggest_file_without_ext}")
+
+    # handle srt files one by one
+    for srt_file in srt_files:
+        if without_extension(srt_file).startswith(biggest_file_without_ext):
+            # already the same start as the biggest file, so skip
+            continue
+        # not the same start, so rename the srt file
+        filename_only = os.path.basename(srt_file)  # like "14_English.srt", so without path
+        filename_only = filename_only.replace("_", ".")  # replace underscore with dot
```

Comment:
Why? This seems unrelated functionality?

Reply:
It makes VLC and thus the user happy ... which is the goal of this PR.

See https://github.com/sabnzbd/sabnzbd/issues/2880#issuecomment-2168680195 : if we keep the "_" in the filename, it won't show up nicely in VLC. If we replace the "_" with ".", it does appear nicely in VLC.

Reply:
I've added a comment to make that clearer